File: clients/ruby/test/client_test.rb
require "minitest/autorun"
require "hhm_client"

class ClientContractTest < Minitest::Test
  def test_health_contract
    observed = nil
    transport = lambda do |method, url, headers, body, timeout|
      observed = [method, url, headers, body, timeout]
      [200, '{"ok":true}']
    end
    client = HhmClient::Client.new(base_url: "https://api.example.com/", token: "secret", transport: transport)
    assert_equal({"ok" => true}, client.health)
    assert_equal :get, observed[0]
    assert_equal "https://api.example.com/healthz", observed[1]
    assert_equal "Bearer secret", observed[2]["authorization"]
  end
end

File: clients/php/tests/client_test.php
<?php

declare(strict_types=1);

require __DIR__ . '/../src/Client.php';

use HackerHouseMedellin\HhmClient\Client;

$observed = null;
$client = new Client('https://api.example.com/', 'secret', function ($method, $url, $headers, $body, $timeout) use (&$observed) {
    $observed = [$method, $url, $headers, $body, $timeout];
    return [200, '{"ok":true}'];
});
$result = $client->health();
assert($result === ['ok' => true]);
assert($observed[0] === 'GET');
assert($observed[1] === 'https://api.example.com/healthz');
assert($observed[2]['authorization'] === 'Bearer secret');
echo "php client contract ok\n";
